flowmap/formats/ugrid.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import pathlib
+
+# TODO, switch to pyugrid after next release
+import netCDF4
+import numpy as np
+
+from tvtk.api import tvtk
+
+from .formats import transform
+from .netcdf import NetCDF
+from .. import particles
+
+logger = logging.getLogger(__name__)
+
+
+class UGrid(NetCDF):
+    """UGrid NetCDF format"""
+
+    def validate(self):
+        """validate a file"""
+        valid = True
+
+        with netCDF4.Dataset(self.path) as ds:
+            variables = ds.variables.keys()
+        for var in ("mesh2d_ucx", "mesh2d_ucy", "mesh2d_face_nodes", "mesh2d_node_x", "mesh2d_node_x"):
```

Comment:
Hopefully those variables are constant (not case sensitive, can also remember nc files without hte 'mesh2d_' part) Shouldn't be a problem within this script?

Reply:
This is hardcoded for now. I plan to switch to pyugrid once my pull requests get merged into the main line (for example the flexible mesh support https://github.com/siggyf/pyugrid/tree/fedor/flexiblemesh)